src/server/trpc.ts

```diff
@@ -36,11 +37,25 @@ const t = initTRPC.context<Context>().create({
   },
 })
 
-/**
- * Create a router
- * @see https://trpc.io/docs/v10/router
- */
-export const router = t.router
+// Setting outer context with TPRC will not get us correct path during request batching, only by setting logger context in
+// the middleware do we get the exact path to log
+const loggerMiddleware = t.middleware(async ({ path, next }) => {
+  const start = Date.now()
+  const logger = createBaseLogger(path)
+  const durationInMs = Date.now() - start
+
+  const result = await next({
+    ctx: { logger },
+  })
+
+  if (result.ok) {
+    logger.info('success', { durationInMs })
+  } else {
+    logger.error('failure', { durationInMs, error: result.error })
+  }
+
+  return result
```

Comment:
@karrui added the `durationInMs` code block from the docs just so every request to a path gets logged

Not logging anything beyond that (some projs have conventions of always logging the req/res payload, but that shouldnt be opted in by default in case of sensitive info)